.github/workflows/pr.yml

```diff
@@ -107,21 +113,35 @@ jobs:
         with:
           fetch-depth: 0
 
+      - name: Update cmd_type as per the mvn_directives
+        if: env.SET_CMD_TYPE == null
+        run: |
+          if [[ -n "${{ inputs.additional_mvn_directives }}" && "${{ inputs.additional_mvn_directives }}" == *"-DrunITs"* ]]; then
+            echo "SET_CMD_TYPE=BUILD,UNIT_TEST,INTEGRATION_TEST" >> $GITHUB_ENV
+          else
+            echo "SET_CMD_TYPE=BUILD,UNIT_TEST" >> $GITHUB_ENV
+          fi
+        shell: bash
+
       - name: Determine which changes occurred
         id: change_detection
         uses: hv-actions/change-detection-builder@stable
 
-      - name: Build & Run unit-tests
+      - name: Build & Run tests
         uses: lumada-common-services/gh-composite-actions@stable
         with:
           command: |
             mvn clean verify -DskipTests=false -Daudit -amd \
-            -pl "${{ format('{0},{1}', inputs.modules_to_always_build_in_addition_to_those_with_changes, steps.change_detection.outputs.changed_modules) }}"
+            -pl "${{ format('{0},{1}', inputs.modules_to_always_build_in_addition_to_those_with_changes, steps.change_detection.outputs.changed_modules) }}" \
+            ${{ inputs.additional_mvn_directives }}
         env:
-          cmd_type: BUILD,UNIT_TEST
+          cmd_type: ${{ env.SET_CMD_TYPE }}
           unit_test_reporter: 'java-junit'
           unit_test_fail_on_error: 'true'
           unit_test_report_path: '**/target/surefire-reports/*.xml'
+          int_test_reporter: 'java-junit'
```

Comment:
What happens if you have the `int_test_*` parameters here but `cmd_type` equals just `BUILD,UNIT_TEST`?

Reply:
It will not break anything. If we have 'int_test_*' environment variables and `cmd_type=BUILD,UNIT_TEST`, it will not execute any integration test steps from the gh-composite-action. As we have a condition in the gh-composite-action, it will only execute the integration test step if `cmd_type` contains `INTEGRATION_TEST`.

https://github.com/pentaho/pentaho-analyzer/actions/runs/8536162424/job/23384194356
For example, here 'additional_mvn_directives' does not contain '-DrunITs'. So, in this case, it is showing a warning message for unit test only and not showing any warning or any other message for integration test, as it is not executing any integration test steps.